src/newrelic/integration/test.go

```diff
@@ -315,6 +316,122 @@ func (t *Test) compareResponseHeaders() {
 	}
 }
 
+// Handling EXPECT_SPAN_EVENTS_LIKE is different than normal payload compare so
+// a different function is required
+func (t *Test) compareSpanEventsLike(harvest *newrelic.Harvest) {
+	// convert array of expected spans JSON to interface representation
+	var x2 interface{}
+
+	if nil == t.spanEventsLike {
+		return
+	}
+
+	if err := json.Unmarshal(t.spanEventsLike, &x2); nil != err {
+		t.Fatal(fmt.Errorf("unable to parse expected spans like json for fuzzy matching: %v", err))
+		return
+	}
+
+	// expected will be represented as an array of "interface{}"
+	// each element will be the internal representation of the JSON
+	// for each expected span
+	// this is needed for the call to isFuzzyMatch Recursive later
+	// when comparing to the actual spans in their internal representation
+	expected := x2.([]interface{})
+
+	// now parse actual span data JSON into interface representation
+	var x1 interface{}
+
+	es := *harvest.SpanEvents
+	id := newrelic.AgentRunID("?? agent run id")
+	actualJSON, err := es.Data(id, time.Now())
+	if nil != err {
+		t.Fatal(fmt.Errorf("unable to access span event JSON data: %v", err))
+		return
+	}
+
+	// scrub actual spans
+	scrubjson := ScrubLineNumbers(actualJSON)
+	scrubjson = ScrubFilename(scrubjson, t.Path)
+	scrubjson = ScrubHost(scrubjson)
+
+	// parse to internal format
+	if err := json.Unmarshal(scrubjson, &x1); nil != err {
+		t.Fatal(fmt.Errorf("unable to parse actual spans like json for fuzzy matching: %v", err))
+		return
+	}
+
+	// expect x1 to be of type "[]interface {}" which wraps the entire span event data
+	// within this generic array there will be:
+	// - a string of the form "?? agent run id"
+	// - a map container "events_seen" and "reservoir_size"
+	// - an array of arrays containing:
+	//    - map containing main span data
+	//    - map for attributes(?)
+	//    - map of CLM data
```

Comment:
Could you add an example of `/*EXPECT_SPAN_EVENTS_LIKE` to the description of the PR?

Reply:
But this comment is related to an actual span events as captured in the harvest by the daemon/integration_runner.  We only keep the final "an array of arrays containing" part as that is the array of spans, the other parts are metadata we don't care about for comparing span events.